config/main.py

```diff
@@ -33,6 +35,70 @@ def run_command(command, display_cmd=False, ignore_error=False):
     if proc.returncode != 0 and not ignore_error:
         sys.exit(proc.returncode)
 
+def interface_alias(interface_name):
+    """Return default interface name if alias name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_dict[port_name]['alias']:
+                return port_name
+        print "Invalid interface {}".format(interface_name)
+        sys.exit(0)
+
+def interface_name_to_alias(interface_name):
+    """Return alias interface name if default name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
```

Comment:
- The interface_name validation is done by pthon click package at the initial phase of the function it throws missing argument error.

```
admin@sonic:~$ sudo config interface shutdown None
Invalid interface None
````

```
admin@sonic:~$ sudo config interface shutdown
Usage: config interface shutdown [OPTIONS] <interface_name>
Error: Missing argument "interface_name".
```

Reply:
I wasn't asking what happens if a user provides the string "None" as the interface name.

My question is regarding the fact that you have an `if` statement (`if interface_name is not None:`) which returns a value but no corresponding `else` statement. Thus, if `interface_name` is of type None (not the string "None"), this function will not return anything. I think this was overlooked, so I was asking for an explanation to see if this is the behavior you expect.

Reply:
I wasn't asking what happens if a user provides the string "None" as the interface name.

My question is regarding the fact that you have an `if` statement (`if interface_name is not None:`) which returns a value but no corresponding `else` statement. Thus, if `interface_name` is of type None (not the string "None"), this function will not return anything. I think this was overlooked, so I was asking for an explanation to see if this is the behavior you expect.

Reply:
Joe I got your point.  When the interface_name variable type is None, even though else is not defined inside the corresponding function the return type of interface_alias_to_name() will be None.  All parent functions  are equipped to handle this case. So, Invalid Interface message will be raised to the user. Even if we add an else the behavior will be the same.

```def shutdown(interface_name, verbose):
    """Shut down interface"""
    if get_interface_mode() == "alias":
        interface_name = interface_alias_to_name(interface_name)
        if interface_name is None:
            print "Invalid interface {}".format(interface_name)<--------
            raise click.Abort()
)
```

```
admin@sonic:~$ sudo config interface shutdown hundredGigE1/1
Invalid interface None
Aborted!
admin@sonic:~$
```

Reply:
I still believe that every path should explicitly return a value. Otherwise the behavior of the function isn't easily determined to the reader. I think you can easily fix this as follows:

```
    if interface_alias is not None:
        for port_name in natsorted(port_dict.keys()):
            if interface_alias == port_dict[port_name]['alias']:
                return port_name
        print "Invalid interface {}".format(interface_alias)
       
    return None
```